cupy/_core/_routines_math.pyx

```diff
@@ -1037,11 +1037,15 @@ _subtract = create_arithmetic(
     cutensor_op=('OP_ADD', 1, -1), scatter_op='sub')
 
 
+# NB: Cannot define loops with short ints in the NEP 50 world. Consider
+# `cupy.arange(3, dtype=cp.uint8) / (-2)`. It would select the 'BB->d' loop,
+# and the kernel would have a declaration `uint8_t in1;`, this converts
+# -2 to uint8_t at initialization (modulo UINT8_MAX, likely).
 _true_divide = create_ufunc(
     'cupy_true_divide',
-    ('bb->d', 'BB->d', 'hh->d', 'HH->d', 'ii->d', 'II->d', 'll->d', 'LL->d',
-     'qq->d', 'QQ->d', 'ee->e', 'ff->f', 'dd->d', 'FF->F', 'DD->D'),
-    'out0 = (out0_type)in0 / (out0_type)in1',
+    ('qq->d', 'QQ->d',
```

Comment:
Right, this has a slight miss for mixed signs, but that seems OK.

In principle it would be nice to solve this with a custom promotion step for the ufunc.  This could be very nice, because for (almost!) all NumPy ufunc, we should be able to at least try and use:
```
dtypes = ufunc.resolve_dtypes()
```